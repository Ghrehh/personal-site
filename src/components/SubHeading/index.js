import React from 'react';
import PropTypes from 'prop-types';

import * as oneOrManyNodesModel from 'models/one-or-many-nodes.js';
import styles from './styles.module.css';

const SubHeading = ({ children, className }) => {
  const combinedClass = `${styles.subHeading} ${className}`;

  return <h2 className={combinedClass}>{children}</h2>;
};

SubHeading.propTypes = {
  children: oneOrManyNodesModel.PropTypes.isRequired,
  className: PropTypes.string
};

export default SubHeading;
