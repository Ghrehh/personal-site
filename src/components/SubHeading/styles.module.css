.subHeading {
  font-size: 24px;
  text-align: center;
  font-weight: 800;
  letter-spacing: 2px;
  margin: auto;
  margin-bottom: 1px;
  font-family: var(--slab-font);
  max-width: 600px;
  line-height: 28px;
}
