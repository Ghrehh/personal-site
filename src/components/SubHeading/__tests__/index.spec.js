import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import SubHeading from '../';

describe('SubHeading', () => {
  let testContext = {};
  let component = (
    <SubHeading className='foo'>
      FooBarBaz
    </SubHeading>
  )

  beforeEach(() => {
    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
