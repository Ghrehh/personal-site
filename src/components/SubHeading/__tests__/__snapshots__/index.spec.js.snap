// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeading renders the component correctly 1`] = `
<h2
  className="subHeading foo"
>
  FooBarBaz
</h2>
`;
